pkg/internal/blockutil.go

```diff
@@ -0,0 +1,237 @@
+package internal
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os/exec"
+	"path/filepath"
+	"strings"
+)
+
+var (
+	ExecCommand          = exec.Command
+	FilePathGlob         = filepath.Glob
+	FilePathEvalSymLinks = filepath.EvalSymlinks
+	mountFile            = "/proc/1/mountinfo"
+)
+
+const (
+	// StateSuspended is a possible value of BlockDevice.State
+	StateSuspended = "suspended"
+	// DiskByIDDir is the path for symlinks to the device by id.
+	DiskByIDDir = "/dev/disk/by-id/"
+)
+
+// BlockDevice is the a block device as output by lsblk.
+// All the fields are lsblk columns.
+type BlockDevice struct {
+	Name   string `json:"name"`
+	KName  string `json:"kname"`
+	Type   string `json:"type"`
+	Model  string `json:"model,omitempty"`
+	Vendor string `json:"vendor,omitempty"`
+	State  string `json:"state,omitempty"`
+	FSType string `json:"fsType"`
+	Size   string `json:"size"`
+	// Children   []BlockDevice `json:"children,omitempty"`
+	Rotational string `json:"rota"`
+	ReadOnly   string `json:"ro,omitempty"`
+	Removable  string `json:"rm,omitempty"`
+	PathByID   string `json:"pathByID,omitempty"`
+	Serial     string `json:"serial,omitempty"`
+	PartLabel  string `json:"partLabel,omitempty"`
+}
+
+// ListBlockDevices using the lsblk command
+func ListBlockDevices() ([]BlockDevice, []string, error) {
+	// var output bytes.Buffer
+	var blockDevices []BlockDevice
+
+	deviceFSMap, err := GetDeviceFSMap()
+	if err != nil {
+		return []BlockDevice{}, []string{}, fmt.Errorf("failed to list block devices: %w", err)
+	}
+
+	columns := "NAME,ROTA,TYPE,SIZE,MODEL,VENDOR,RO,RM,STATE,KNAME,SERIAL,PARTLABEL"
+	args := []string{"--pairs", "-b", "-o", columns}
```

Comment:
can use `lsblk --json` and marshal the output in the BlockDevices struct.  This will take time to test. So I'll update this method in  follow up PR

Reply:
Sounds good.